Lib/dataclasses.py

```diff
@@ -1530,7 +1530,11 @@ class C(Base):
     for item in fields:
         if isinstance(item, str):
             name = item
-            tp = 'typing.Any'
+            typing = sys.modules.get('typing')
+            if typing:
+                tp = typing.Any
```

Comment:
Could we use `"__import__('typing').Any"`? Since `get_type_hints()` calls `eval()`, that should work regardless of whether the dataclass is defined in a place where typing is imported.

I can provide a more robust solution on 3.14 using `__annotate__`.

Reply:
Nikita's current solution feels in keeping with what `dataclasses` does elsewhere: https://github.com/python/cpython/blob/e9681211b9ad11d1c1f471c43bc57cac46814779/Lib/dataclasses.py#L809-L815

`dataclasses` in general takes great pains to avoid importing `typing` if it's not already imported. I think these days `typing` is actually a faster import than `dataclasses`, but for now I think it's probably best to stick with the existing general philosophy of `dataclasses` there.

Reply:
My solution would not involve importing typing in the dataclasses module.

Reply:
Oh I see! That's very clever, but I think it would result in some pretty funky output if the user just tried to access `__annotations__` directly (or via `inspect.get_annotations`) rather than via `get_type_hints`, no?

Reply:
> I can provide a more robust solution on 3.14 using __annotate__.

I am already working on that for 3.14+ 😉

Reply:
> Could we use "__import__('typing').Any"?

I feel like this might not be a good idea for older versions. Since right now `dataclasses` do not import `typing` directly, this might cause performance regressions. This PR can be easily backported.

Plus, we have `__annotate__` for the future versions.

Looks like a win-win to me.

Reply:
@AlexWaygood it would look surprising, yes, but it would solve the user-facing issue that got reported.

Reply:
I agree with @AlexWaygood, this might be a very scary thing to see for annotations without the evaluation 😱

Reply:
That's fair, but then I don't think we should go with the current solution either. My example below seems like a reasonable scenario for how this could end up getting used in practice: user code does not import `typing` but calls `make_dataclass()`, later something like cattrs run and tries to get the type hints for the class.

With your PR, this will work only if `typing` happened to have been imported at the time the dataclass was created. But that becomes very fragile: it means that working code can break if you refactor your code to reorder imports, or remove an `import typing` from an unrelated part of the code.

Reply:
Ok, let's try both: `typing.Any` if `typing` exists, and `"__import__('typing').Any"` if it doesn't 👍